src/server/server.rs

```diff
@@ -179,11 +181,7 @@ where
         let trans = ServerTransport::new(raft_client);
         health_service.set_serving_status("", ServingStatus::NotServing);
 
-        let tiflash_engine = cfg
-            .value()
-            .labels
-            .iter()
-            .any(|entry| entry.0 == ENGINE && entry.1 == TIFLASH);
+        let is_tiflash_engine = contain_tiflash_engine_label(&cfg.value().labels);
```

Comment:
TiFlash depends on TiKV instead of the opposite. We should use switch to enable or disable certain feature instead of doing something special for tiflash only.

Reply:
@overvenus  As Jay said, it will be better to add a general switch to control this behavior and tiflash will enable this switch.